src/priority_queue.js

```diff
@@ -0,0 +1,38 @@
+class PriorityQueue {
```

Comment:
Is this a work in progress?  It is not honoring priority (looks like just a copy and paste of `Queue`?).

In order to get more meaningful feedback, I suggest only submitting PRs for code that is completed and ready for review.  One approach for this project might be to implement each data structure separately, and submit a PR for each of those data structures (with tests).